File: src/Plotly.NET/Traces/ObjectAbstractions/Sunburst.fs
namespace Plotly.NET.TraceObjects

open Plotly.NET
open Plotly.NET.LayoutObjects
open DynamicObj
open System
open System.Runtime.InteropServices


type SunburstRoot() =
    inherit DynamicObj()

    static member init(?Color: Color) =

        SunburstRoot() |> SunburstRoot.style (?Color = Color)

    static member style(?Color: Color) =
        fun (root: SunburstRoot) ->
            root
            |> DynObj.withOptionalProperty "color" Color

type SunburstLeaf() =
    inherit DynamicObj()

    static member init(?Opacity: float) =

        SunburstLeaf() |> SunburstLeaf.style (?Opacity = Opacity)

    static member style(?Opacity: float) =
        fun (leaf: SunburstLeaf) ->

            leaf
            |> DynObj.withOptionalProperty "opacity" Opacity

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/Sunburst.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.SunburstRoot InitSunburstRoot
        (
            Optional<Color> Color = default
        ) 
            =>
                Plotly.NET.TraceObjects.SunburstRoot.init(
                    Color: Color.ToOption()
                );
    public static Plotly.NET.TraceObjects.SunburstLeaf InitSunburstLeaf
        (
            Optional<double> Opacity = default
        ) 
            =>
                Plotly.NET.TraceObjects.SunburstLeaf.init(
                    Opacity: Opacity.ToOption()
                );
}

